pkg/stub/handler.go

```diff
@@ -372,10 +472,29 @@ func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
 		if srcfg.Name != v1alpha1.SamplesResourceName {
 			return nil
 		}
+
+		// pattern is 1) come in with delete timestample, event delete flag false
+		// 2) then after we remove finalizer, comes in with delete timestamp
+		// and event delete flag true
 		if event.Deleted {
-			logrus.Println("SampleResource deleted")
-			h.samplesResource = nil
-			return h.CreateDefaultResourceIfNeeded()
+			// possibly tell poller to stop
+			logrus.Info("A previous delete attempt has been successfully completed")
+			return nil
+		}
+		if srcfg.DeletionTimestamp != nil {
+			if h.NeedsFinalizing(srcfg) {
+				logrus.Println("Initiating finalizer processing for a SampleResource delete attempt")
```

Comment:
strictly speaking, finalizing should also remove the imagestreams, templates, and secrets we created in the openshift namespace.  (but only the ones we created/own).

Reply:
By "created/own", does that mean existing ones we've updated ?

Reply:
yes.  Anything that the samplesoperator knows how to create and which isn't in the Skipped list.

I think the best thing to do would be for the operator to put a label or annotation on the things it creates.  If someone later adds one of those items to the Skipped list, we should remove the label/annotation.

Reply:
(if we weren't going cross-namespace, i'd suggest an ownerref and rely on GC)